include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -5707,13 +5720,152 @@ FEEvaluationAccess<dim, dim, Number, is_face, VectorizedArrayType>::operator=(
 }
 
 
+template <int dim, typename Number, bool is_face, typename VectorizedArrayType>
+inline DEAL_II_ALWAYS_INLINE Tensor<1, dim, VectorizedArrayType>
+FEEvaluationAccess<dim, dim, Number, is_face, VectorizedArrayType>::get_value(
+  const unsigned int q_point) const
+{
+  // Check if Piola transform is required
+  if (this->data->element_type ==
+      internal::MatrixFreeFunctions::ElementType::tensor_raviart_thomas)
+    {
+#  ifdef DEBUG
+      Assert(this->values_quad_initialized == true,
+             internal::ExcAccessToUninitializedField());
+#  endif
+
+      AssertIndexRange(q_point, this->n_quadrature_points);
+      Assert(this->J_value != nullptr,
+             internal::ExcMatrixFreeAccessToUninitializedMappingField(
+               "update_values"));
+      const std::size_t                   nqp = this->n_quadrature_points;
+      Tensor<1, dim, VectorizedArrayType> value_out;
+
+      // Cartesian cell
+      if (!is_face &&
+          this->cell_type == internal::MatrixFreeFunctions::cartesian)
+        {
+          const Tensor<2, dim, dealii::VectorizedArray<Number>> jac =
+            this->jacobian[1];
+          const VectorizedArrayType inv_det = determinant(this->jacobian[0]);
```

Comment:
Oh yes, sorry for the confusion. The division was instead of previously computing the inverse of `this->jacobian[0]`. But yes, maybe the product of the diagonal elements is faster than computing the determinant. What do you think?

Reply:
Yes indeed, the product of the diagonals is faster (2 multiplications in 3D) compared to the determinant, which does 4 multiplications, 5 FMAs in 3D according to https://github.com/dealii/dealii/blob/82cf0e0296706aaec7b9502b93bfb032fd385e44/include/deal.II/base/tensor.h#L2749-L2756